webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingValueCell.js

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Flex, FlexItem } from '@patternfly/react-core';
+import { PencilAltIcon } from '@patternfly/react-icons';
+import SettingValueEdit from './SettingValueEdit';
+import SettingValue from './SettingValue';
+import { checkEncryptedValue } from '../SettingsTableHelpers';
+
+const SettingValueCell = ({ setting, index }) => {
+  const [editingRow, setEditingRow] = useState(-1);
+  const [settingData, setSettingData] = useState(setting);
+  const updateSetting = newValue => {
+    const newSetting = { ...setting, value: newValue };
+    setSettingData({ ...settingData, value: checkEncryptedValue(newSetting) });
+    setEditingRow(-1);
+  };
+
+  const isEditing = () => index === editingRow;
+  return (
+    <>
+      {isEditing() ? (
+        <SettingValueEdit setting={settingData} updateSetting={updateSetting} />
+      ) : (
+        <Flex
+          justifyContent={{ default: 'justifyContentSpaceBetween' }}
+          alignItems={{ default: 'alignItemsCenter' }}
+          flexWrap={{ default: 'nowrap' }}
+        >
+          <FlexItem>
+            <SettingValue setting={settingData} />
+          </FlexItem>
+          <FlexItem>
+            <Button
+              onClick={() => setEditingRow(index)}
```

Comment:
Could `EditingRow` just be a boolean and replace `const isEditing = () => index === editingRow;` ?